python/cudf/cudf/core/column/__init__.py

```diff
@@ -23,6 +23,7 @@
     serialize_columns,
 )
 from cudf.core.column.datetime import DatetimeColumn  # noqa: F401
+from cudf.core.column.datetime import DatetimeTZColumn  # noqa: F401
```

Comment:
FYI: you can fix these "unused import" errors by defining `__all__`. If it's in `__all__`, it's "part of the API" of this module and thus flake8 sees it as used.

Reply:
Isn't this more `explicit` though? I always thought the only "official" use for `__all__` was to define what happens when you do `from module import *`

Reply:
I think that defining your module’s public API (rather than not defining it) is the most `explicit` approach. I’m pro-`__all__` in Python libraries.

Reply:
>  think that defining your module’s public API (rather than not defining it) is the most `explicit` approach. 

Hmm, but `__all__` doesn't do that AFAICT. Placing a name in `__all__` doesn't preclude it from appearing in e.g., tab completion options (naming it with a leading `_` does). 

Are there other tools than `flake8` that do something with `__all__`?

Reply:
I think type checkers (mypy, pyright) will utilize `__all__` to see what APIs are "public"

Reply:
Also, to your question about tab completion in IPython/etc., that's discussed here: https://github.com/ipython/ipykernel/issues/129#issuecomment-216870597

The community seems split on the issue, but some folks seem to think limiting to `__all__` by default would be good. Either way -- it seems that `__all__` is viewed as a source of truth for public APIs, and the real question is whether autocompletion should only show public APIs.

> It's configurable, with the option IPCompleter.limit_to__all__ ([config options docs](http://ipython.readthedocs.io/en/stable/config/options/terminal.html)).
>
> The reason we do this is that we often want to use IPython to poke about and debug modules, so we like to see what's really there, not just what's whitelisted as public API. We should, however, do a better job of prioritising completions, so that things in `__all__` come before things that aren't in it.

Reply:
Thanks, both! I'm on board with defining `__all__`

Reply:
I wonder if we can get a linter to enforce that `__all__` is defined (even if empty) always.